### PR TITLE
Refactoring + separated modules

### PR DESCRIPTION
**Pull request by VEndymionV**
Created separated modules - modules for remote (retrofit) and for viewmodel

### CODE REVIEW

File: app/src/main/java/com/example/training1/movies/model/MoviesRepository.kt [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import com.example.training1.movies.viewmodel.model.SearchResult
+import io.reactivex.Single
+import javax.inject.Inject
+
+class MoviesRepository @Inject constructor(
+    private val retrofitMoviesService: MoviesService,
+    private val apiKey: String
+) {
+
+    fun searchForMovies(title: String, type: String? = null): Single<SearchResult> {
```
Comment (Bartosz-Kozajda) [inline app/src/main/java/com/example/training1/movies/model/MoviesRepository.kt]: I think this method should return just List<Movie>

File: app/src/main/java/com/example/training1/movies/model/MoviesRepository.kt [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        return getRemoteData(title, type).map {
+            SearchResult(mapRemoteMovies(it.search), it.totalResults, it.response)
+        }
+    }
+
+    private fun getRemoteData(title: String, type: String? = null): Single<RemoteSearchResult> {
+        return retrofitMoviesService.searchForMovies(apiKey, title, type)
+    }
+
+    private fun mapRemoteMovies(remoteMovie: List<RemoteMovie>): List<Movie> {
```
Comment (Bartosz-Kozajda) [inline app/src/main/java/com/example/training1/movies/model/MoviesRepository.kt]: you should create additional mapper class or write extension function. Repository is not a good place to keep mappers

File: app/src/main/java/com/example/training1/movies/model/MoviesRepository.kt (hunk 1 above)
Comment (VEndymionV) [reply app/src/main/java/com/example/training1/movies/model/MoviesRepository.kt]: Changed in: https://github.com/VEndymionV/Training1/pull/36/commits/7efc6d3467fb39f184539901534330623d16a198

File: app/src/main/java/com/example/training1/movies/model/MoviesRepository.kt (hunk 2 above)
Comment (VEndymionV) [reply app/src/main/java/com/example/training1/movies/model/MoviesRepository.kt]: Changed in: https://github.com/VEndymionV/Training1/pull/36/commits/7efc6d3467fb39f184539901534330623d16a198

**Pull request merged**